### PR TITLE
Delete survey functionality!!! 🗑️

### PR DESCRIPTION
**Pull request by sdevalapurkar**
# Overview

## This PR contains the following changes

- Delete survey functionality!!! 🗑️
- Also wipe attachments associated to survey from s3
- Deal with observations and occurrences later (block observation table does get wiped, but detailed deletion of records under survey level will be dealt with later on)

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)

## How Has This Been Tested?

- Locally

### CODE REVIEW

**Review by cgarrettjones — DISMISSED** [review]

**Review by anissa-agahchen — COMMENTED** [review]
Looks good.

File: api/src/paths/project/{projectId}/survey/{surveyId}/delete.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+       * Get the attachment S3 keys for all attachments associated to this survey
+       * Used to delete them from S3 separately later
+       */
+      const getSurveyAttachmentSQLStatement = getSurveyAttachmentsSQL(Number(req.params.surveyId));
+
+      if (!getSurveyAttachmentSQLStatement) {
+        throw new HTTP400('Failed to build SQL get statement');
+      }
+
+      await connection.open();
```
Comment (NickPhura) [inline api/src/paths/project/{projectId}/survey/{surveyId}/delete.ts:86]: Duplicate open call on line 73

File: api/src/paths/project/{projectId}/survey/{surveyId}/delete.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    defaultLog.debug({ label: 'Delete survey', message: 'params', req_params: req.params });
+
+    if (!req.params.surveyId) {
+      throw new HTTP400('Missing required path param `surveyId`');
+    }
+
+    const connection = getDBConnection(req['keycloak_token']);
+
+    try {
+      await connection.open();
```
Comment (NickPhura) [inline api/src/paths/project/{projectId}/survey/{surveyId}/delete.ts]:
```suggestion
```

Comment (NickPhura) [reply api/src/paths/project/{projectId}/survey/{surveyId}/delete.ts]: Already called on line  86

**Comment by anissa-agahchen** [comment]
Not specific to your PR, but good time to add export NODE_ENV=local to the database Readme

**Comment by anissa-agahchen** [comment]
Doesn't look like there is a rush on this functionality, so might as well add the unit tests

**Review by anissa-agahchen — APPROVED** [review]
Looks good!

**Pull request merged**